docs/design/flow-traffic-mirroring/flow-traffic-mirroring.md

```diff
@@ -0,0 +1,172 @@
+# Flow traffic mirroring
+
+## Idea
+There must be an ability to mirror traffic on a given flow and send it to a specific switch+port+vlan.
+
+## Model
+FlowPath has a list of FlowMirrorPoints, which will contain MirrorGroupId for multicasting on the switch. 
+FlowMirrorPoint itself has a list of MirrorFlowPaths. MirrorFlowPaths contains the endpoint 
+where the traffic should be mirrored and other necessary information to build the path.
+
+![DB model](./model.png "DB model")
+
+If the receiver point is on the mirroring source switch, then no segments will be built for such a MirrorFlowPath, 
+and we will work with it like the paths in one switch flow.
+
+## API
+* Create mirror point:
+  
+  `PUT /flows/{flow_id}/mirror`
+  
+  payload:
+  ```
+  {
+      "mirror_point_id": string,
+      "flow_id": string,
+      "mirror_direction": string [FORWARD|REVERSE],
+      "mirror_point_switch_id": string,
+      "receiver_point": {
```

Comment:
Can we name it "receiver_endpoint"? Because used fields set is equal to our existing data structure named encpoint (so we can reuse it and it's tooling internally). Also I term "receiver" here looks some incorrect / ambiguous. I would like fo search some other term - "sink" (sink_endpoint) for example - it is not perferct, but better. We need some term that unambiguously define egress nature of this endpoint.

Reply:
I like your version. Renamed.